src/classes/context.ts

```diff
@@ -9,15 +9,15 @@ export class Context<T extends {} = any> {
   }
 
   public as<U extends T>(): Context<U> {
-    return this as Context<U>;
+    return this as unknown as Context<U>;
```

Comment:
Really necessary ? 🤔

Reply:
Yeah, there might be another way to constrain `U` somehow, but if you leave it out, the compiler will explicitly tell you to cast it to unknown first.

Reply:
Yeah actually the whole types can now be REALLY simplified thanks to `infer`. Redoing them :)